src/line_agg.rs

```diff
@@ -753,7 +782,7 @@ mod tests {
 
         assert_results(
             results.await.unwrap(),
-            &[(expected.as_str(), lines.len() - 1)],
+            &[(expected.as_str(), lines.len() - 1, None)],
```

Comment:
Looks like this test case is failing..the assertion should have zero as the initial context, and `Some(lines.len() - 1)` as the last_context.